refactor(dev-infra): allow use-case and confusing types to be marked as triaged

atscott opened the pull request:
Some issue reports don't really fall into any of the current buckets that count
towards triage level 2: bug/fix, feature, or refactor. Some reports are:
* working as intended but confusing - the labels might be 'type: confusing', 'comp: docs', 'comp: router'
* generally working as originally designed but a use-case could be argued for a different implementation.
 This type of report is a little hard to triage; it may be neither a bug, nor feature but more of a
 'type: use-case'. These may eventually turn into a bug/fix or feature, but can't necessarily be
 put in those buckets immediately.

atscott committed:
refactor(dev-infra): allow use-case and confusing types to be marked as 'triaged'

Some issue reports don't really fall into any of the current buckets that count
towards triage level 2: bug/fix, feature, or refactor. Some reports are:
* working as intended but confusing - the labels might be 'type: confusing', 'comp: docs', 'comp: router'
* generally working as originally designed but a use-case could be argued for a different implementation.
 This type of report is a little hard to triage; it may be neither a bug, nor feature but more of a
 'type: use-case'. These may eventually turn into a bug/fix or feature, but can't necessarily be
 put in those buckets immediately.

AndrewKushnir commented on .github/angular-robot.yml:
```diff
@@ -154,6 +154,14 @@ triage:
       - "comp: *"
+    -
+      - "type: confusing"
+      - "freq*"
```
I'd propose to avoid treating the `freq` label as required (similar to `refactor` or `feature`).

atscott committed:
fixup! refactor(dev-infra): allow use-case and confusing types to be marked as 'triaged'

pkozlowski-opensource reviewed (approved):
Yay! LGTM, thnx for sending this PR (I should have done it before).
Since we are at it - could you please also update https://github.com/angular/angular/blob/master/docs/TRIAGE_AND_LABELS.md to reflect those new types (hence adding the cleanup label)

atscott committed:
refactor(dev-infra): Update triage labels documentation

Add new type: confusing and type: use-case labels to the triage readme as well
as clarify that freq and severity are only required for type: bug/fix

atscott commented:
> Since we are at it - could you please also update https://github.com/angular/angular/blob/master/docs/TRIAGE_AND_LABELS.md to reflect those new types (hence adding the cleanup label)

👍  Done - good addition. I didn't even know that doc existed 😄

josephperrott reviewed (approved):
LGTM
